test/test_nested_tensor_masking.py

```diff
@@ -1,350 +1,976 @@
         ])
-        
-        expected_nt0 = torch.tensor([[[[[1, 2],
-                                        [3, 4]]]],
-                                     [[[[5, 6],
-                                        [7, 8]]]]])
 
-        expected_nt1 = nt.nested_tensor([
-            torch.tensor([[[[1, 2],
-                            [3, 4]]]]),
-            torch.tensor([[[[5, 6],
-                            [7, 8]]]]),
-        ])
+        tensor = torch.tensor([[[1, 2, 3]],
+                               [[4, 0, 0]]])
+        mask = torch.tensor([[[ True,  True,  True]],
+                             [[ True, False, False]]])
+
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=1)
+        TestCase.assertEqual(self, expected_nt1, res_nt)
 
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=2)
+        TestCase.assertEqual(self, expected_nt2, res_nt)
+
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=3)
+        TestCase.assertEqual(self, expected_nt3, res_nt)
+
+        self.assertRaises(RuntimeError, lambda: nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=4))
+
+    def test_ntftm_test_multi_tensor_mix_mask3(self):
         expected_nt2 = nt.nested_tensor([
             nt.nested_tensor([
-                torch.tensor([[[1, 2],
+                torch.tensor([[[1, 2, 3, 4],
+                               [5, 6, 7, 8]]]),
+                torch.tensor([[[0, 0],
                                [3, 4]]]),
-            ]),
+                torch.tensor([[[1]]])
+            ])
+        ])
+
+        expected_nt3 = nt.nested_tensor([
+                nt.nested_tensor([
+                    nt.nested_tensor([
+                        torch.tensor([[1, 2, 3, 4],
+                                      [5, 6, 7, 8]])
+                    ]),
+                    nt.nested_tensor([
+                        torch.tensor([[0, 0], 
+                                      [3, 4]])
+                    ]),
+                    nt.nested_tensor([
+                        torch.tensor([[1]])
+                    ]),
+                ])
+            ])
+
+        expected_nt4 = nt.nested_tensor([
             nt.nested_tensor([
-                torch.tensor([[[5, 6],
-                               [7, 8]]]),
-            ]),
+                nt.nested_tensor([
+                    nt.nested_tensor([
+                        torch.tensor([1, 2, 3, 4]),
+                        torch.tensor([5, 6, 7, 8])
+                    ])
+                ]),
+                nt.nested_tensor([
+                        nt.nested_tensor([
+                            torch.tensor([0, 0]),
+                            torch.tensor([3, 4])
+                        ])
+                ]),
+                nt.nested_tensor([
+                        nt.nested_tensor([
+                            torch.tensor([1]),
+                            torch.tensor([], dtype=torch.long)
```

Comment:
Let's talk about why you'd want this

Reply:
will follow up once to_nt is moved to C++